src/main/java/nl/tudelft/contextproject/gui/PreviewController.java

```diff
@@ -40,133 +42,220 @@
     @FXML private Button play;
     @FXML private Button stop;
     @FXML private Button back;
+    @FXML private Button confirm;
     @FXML private ChoiceBox<Shot> cb;
     @FXML private TextField duration;
     @FXML private Rectangle highlight1;
     @FXML private Rectangle highlight2;
     @FXML private Rectangle highlight3;
     @FXML private Rectangle highlight4;
-    
-    private int currentShot;
+    @FXML private Label error;
+
+    private SimpleIntegerProperty currentShot;
+    private int currentSelected;
     private Timeline timeline;
     private ArrayList<Shot> shots;
```

Comment:
"Program to an interface, not to an implementation". Or something similar, as stated in the Software Engineering Methods course from mr. Bachelli. Anyway, you should change this declaration to List<Shot> instead of ArrayList<Shot>